apps/web/src/app/arrangementer/page.tsx

```diff
@@ -1,76 +1,125 @@
 "use client"
 
-import { EventsViewToggle } from "@/components/molecules/EventsViewToggle"
+import { EventCalendar } from "@/components/organisms/EventCalendar"
+import { CalendarNavigation } from "@/components/organisms/EventCalendar/CalendarNavigation"
 import { useTRPC } from "@/utils/trpc/client"
 import type { EventFilterQuery } from "@dotkomonline/types"
-import { Title } from "@dotkomonline/ui"
+import { Icon, Tabs, TabsContent, TabsList, TabsTrigger, Title } from "@dotkomonline/ui"
 import { getCurrentUTC } from "@dotkomonline/utils"
 import { useQuery } from "@tanstack/react-query"
 import { roundToNearestMinutes } from "date-fns"
+import { useRouter, useSearchParams } from "next/navigation"
 import { useState } from "react"
 import { EventFilters } from "./components/EventFilters"
 import { EventList, EventListSkeleton } from "./components/EventList"
 import { useEventAllInfiniteQuery, useEventAllQuery } from "./components/queries"
 
 const EventPage = () => {
+  const router = useRouter()
+  const searchParams = useSearchParams()
   const now = roundToNearestMinutes(getCurrentUTC(), { roundingMethod: "floor" })
   const [filter, setFilter] = useState<EventFilterQuery>({})
 
+  const view = searchParams.get("view") || "list"
+  const year = Number.parseInt(searchParams.get("y") || now.getFullYear().toString())
+  const month = Number.parseInt(searchParams.get("m") || (now.getMonth() + 1).toString()) - 1 // Convert to 0-based month
+
   const trpc = useTRPC()
   const { data: isStaff = false } = useQuery(trpc.user.isStaff.queryOptions())
 
   const { eventDetails: futureEventWithAttendances, isLoading } = useEventAllQuery({
     filter: {
       ...filter,
-
       byEndDate: {
         max: null,
         min: now,
       },
-
       excludingOrganizingGroup: ["velkom"],
       excludingType: isStaff ? [] : undefined,
       orderBy: "asc",
     },
     page: {
-      // There should never be a lot of future events, so we fetch them all here to support different client-side groupings
       take: 1000,
     },
   })
 
   const { eventDetails: pastEventWithAttendances, fetchNextPage } = useEventAllInfiniteQuery({
     filter: {
       ...filter,
-
       byEndDate: {
         max: now,
         min: null,
       },
-
       excludingOrganizingGroup: ["velkom"],
       excludingType: isStaff ? [] : undefined,
       orderBy: "desc",
     },
   })
 
+  const handleViewChange = (newView: string) => {
+    const params = new URLSearchParams(searchParams.toString())
+
+    if (newView === "list") {
+      router.push("/arrangementer")
+    } else {
+      params.set("view", newView)
+      router.push(`/arrangementer?${params.toString()}`)
```

Comment:
This should probably not pollute the user's history
```suggestion
      router.replace("/arrangementer")
    } else {
      params.set("view", newView)
      router.replace(`/arrangementer?${params.toString()}`)
```